homeassistant/components/fan/demo.py

```diff
@@ -0,0 +1,67 @@
+"""
+Demo garage door platform that has a fake fan.
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/demo/
+"""
+
+from homeassistant.components.fan import (SPEED_LOW, SPEED_MED, SPEED_HIGH,
+                                          FanEntity)
+from homeassistant.const import STATE_OFF
+
+
+FAN_NAME = 'Living Room Fan'
+FAN_ENTITY_ID = 'fan.living_room_fan'
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices_callback, discovery_info=None):
+    """Setup demo garage door platform."""
+    add_devices_callback([
+        DemoFan(hass, FAN_NAME, STATE_OFF),
+    ])
+
+
+class DemoFan(FanEntity):
+    """A demonstration fan component."""
+
+    def __init__(self, hass, name: str, initial_state: str) -> None:
+        """Initialize the entity."""
+        self.hass = hass
+        self.speed = initial_state
+        self.is_oscillating = False
+        self._name = name
+
+    @property
+    def name(self) -> str:
+        """Get entity name."""
+        return self._name
+
+    @property
+    def should_poll(self):
+        """No polling needed for a demo fan."""
+        return False
+
+    @property
+    def speed_list(self) -> list:
+        """Get the list of available speeds."""
+        return [STATE_OFF, SPEED_LOW, SPEED_MED, SPEED_HIGH]
```

Comment:
Well, my original plan was to support an integer 0-100 or set speeds and @balloob recommended we follow the fan from the climate component - I agree.

If we get 0-100% fan platforms, either they can set a % for each level, or they can update the component to accept integer as well.